client/error.go

```diff
@@ -0,0 +1,9 @@
+package client
+
+import (
+	"github.com/giantswarm/microerror"
+)
+
+var invalidConfigError = &microerror.Error{
```

Comment:
Matcher `IsInvalidConfig(err error)` missing on purpose?

Reply:
Yup, I was adding stuff as they were needed.

But now when you mention it, it looks like I should add it and use `nodes.IsInvalidConfig(err error)` instread of `masters.IsInvalidConfig(err error)` and `instance.IsInvalidConfig(err error)` in few places.

P.S. I'm not sure I like this repetition of `error.go` and practically same errors/functions in it all over the project, it seems to be error prone.

Reply:
> But now when you mention it, it looks like I should add it and use nodes.IsInvalidConfig(err error) instead of masters.IsInvalidConfig(err error) and instance.IsInvalidConfig(err error) in few places.

Or not :thinking:, ok I have to check first how is error comparison working in `IsInvalidConfig(err error)`.

Reply:
Since the specific instance of error type differs from package to another, the matcher must be accompanied with the specific types in every package.

Reply:
So currently there is no need anywhere for `client.IsInvalidConfig(err)`, the factory is created in `resource_set.go` (it will be changed after rebasing on `master`), and the error from `client.NewFactory(config)` is then just propagated all the way back to `main.go` where it panics.

Reply:
But this should be corrected in other places in the other pull request for refactoring `masters` and `instance`.